_scripts/docker-compose.yml

```diff
@@ -0,0 +1,45 @@
+# Example command:
+#
+#   DOCKER_IMAGE=humble-source docker-compose run gpu
+
+x-base-cpu: &base-cpu
+  privileged: true
+  network_mode: host
+  environment:
+    QT_X11_NO_MITSHM: 1
+  volumes:
+    - /tmp/.X11-unix:/tmp/.X11-unix
+    - ~/.Xauthority:/root/.Xauthority
+  command: /bin/bash
+
+x-base-gpu: &base-gpu
+  <<: *base-cpu
+  deploy:
+    resources:
+      reservations:
+        devices:
+          - driver: nvidia
+            count: 1
+            capabilities: [gpu]
+  environment:
+    NVIDIA_VISIBLE_DEVICES: all
+    NVIDIA_DRIVER_CAPABILITIES: all
+
+services:
+  cpu:
+    <<: *base-cpu
+    image: moveit/moveit2:$DOCKER_IMAGE
+    volumes:
+      - moveit2-data:/root
+    environment:
+      DISPLAY: $DISPLAY
+  gpu:
+    <<: *base-gpu
+    image: moveit/moveit2:$DOCKER_IMAGE
+    volumes:
+      - moveit2-data:/root
+    environment:
+      DISPLAY: $DISPLAY
```

Comment:
@smartin015: I'm wondering if this works as intended.

According to the YAML spec and my testing, this does not override/merge `environment` into the `environment` map specified in `x-base-gpu`, which doesn't override/merge the `environment` map specified in `x-base-cpu`.

I've had to do something like this to get it to work:

```yaml
x-base-cpu: &base-cpu
  ...
  environment: &base-cpu-env
    QT_X11_NO_MITSHM: 1
  ...

x-base-gpu: &base-gpu
  <<: *base-cpu
  ...
  environment: &base-gpu-env
    <<: *base-cpu-env
    NVIDIA_VISIBLE_DEVICES: all
    NVIDIA_DRIVER_CAPABILITIES: all

services:
  ...
  gpu:
    <<: *base-gpu
    ...
    environment:
      <<: *base-gpu-env
      DISPLAY: $DISPLAY
```

could you perhaps verify `QT_X11_NO_MITSHM` is being set in the `gpu` container @smartin015? `env | sort` didn't show it for me, unless I made the above changes.

Reply:
My bad on testing somewhat I may have misused some settings while testing. I can confirm what @gavanderhoorn says here. The docker container launches fine but the display isn't working. This fix seems to set the correct environment variables as suggested, but unfortunately, the display wasn't working for me (ie: xeyes). I'm going to revert this change in a day if not resolved to keep non working stuff away from main.

Reply:
Apologies for the breakage - I opened #439 to remove the `x-*` templating since it may be a bit suspect. However, please do revert the change if you'd rather not try to fix forward. Better to have working old instructions than non-working new ones :)

From a clean boot of my Ubuntu 22.04 computer, I ran 

```
DOCKER_IMAGE=humble-source docker-compose run gpu /bin/bash
source install/setup.bash && ros2 run rviz2 rviz2
```

Running `xhost` command shows only `SI:localuser:<my username>`, so it seems that xhost isn't configured in any way beyond the defaults.